catkin_ws/src/barracuda_vision/scripts/yolo_image_listener.py

```diff
@@ -0,0 +1,130 @@
+        confidence = detections.confidence
+        class_id = [pred["class"] for pred in predictions]
+
+    boundingBoxes = BoundingBoxes()
+    boundingBoxes.header = data.header
+    boundingBoxes.image_header = data.header
+    boundingBoxes.bounding_boxes = []
+
+    for i in range(len(detections)):
+        boundingBox = BoundingBox()
+        boundingBox.xmin = int(xyxy[i][0])
+        boundingBox.ymin = int(xyxy[i][1])
+        boundingBox.xmax = int(xyxy[i][2])
+        boundingBox.ymax = int(xyxy[i][3])
+        boundingBox.probability = confidence[i]
+        boundingBox.Class = class_id[i]
+        boundingBoxes.bounding_boxes.append(boundingBox)
+
+    # Annotate and publish detection results
+    box_annotator = sv.BoxAnnotator()
+    annotated_frame = box_annotator.annotate(
+        scene=image.copy(),
+        detections=detections
+    )
+
+    detectionImage = Image()
+    detectionImage.header = data.header
+    detectionImage.encoding = "bgr8"
+    detectionImage.height = image.shape[0]
+    detectionImage.width = image.shape[1]
+    detectionImage.step = image.shape[1] * 3
+
+    detectionImage = cvBridge.cv2_to_imgmsg(annotated_frame, encoding="bgr8")
+
+    pub_object_detector.publish(len(detections))
+    pub_bounding_boxes.publish(boundingBoxes)
+    pub_detection_image.publish(detectionImage)
+
+def listener():
+    """Initialize ROS node and publishers, and subscribe to image topic."""
+    rospy.init_node('yolo_image_listener', anonymous=True)
+    use_local_model = rospy.get_param('~use_local_model', True)  # Default to local model
+    rospy.Subscriber("yolo_input_image", Image, lambda data: infer(data, use_local_model))
+
+    global pub_object_detector, pub_bounding_boxes, pub_detection_image
+    pub_object_detector = rospy.Publisher('object_detector', Int8, queue_size=10)
+    pub_bounding_boxes = rospy.Publisher('bounding_boxes', BoundingBoxes, queue_size=10)
+    pub_detection_image = rospy.Publisher('detection_image', Image, queue_size=10)
+
+    rospy.spin()
+
+if __name__ == '__main__':
+    # Find the localModels directory relative to this script
+    script_dir = os.path.dirname(os.path.abspath(__file__))
+    models_dir = os.path.join(script_dir, '..', 'localModels')
+
+    # Ensure exactly one model file exists in localModels
+    model_files = [f for f in os.listdir(models_dir) if os.path.isfile(os.path.join(models_dir, f))]
+    if not model_files:
+        rospy.logerr(f"No model file found in {models_dir}. Shutting down container.")
+        sys.exit(1)
+    if len(model_files) > 1:
+        rospy.logerr(f"Multiple model files found in {models_dir}. Only one model should be present. Shutting down container.")
+        sys.exit(1)
+
+    # Load the only model file found
+    model_path = os.path.join(models_dir, model_files[0])
+    rospy.loginfo(f"Using model file: {model_path}")
+
+    model = YOLO(model_path)
+    model.eval()
+    preprocess = transforms.Compose([
```

Comment:
yup. lets remove the non-local model capability from this code completely and just focus on the single function of using the local model. that is the idea I want to transition into. So you may need to clean up a lot of code, this file should be way smaller than it is.